### PR TITLE
feat: implement 'argocd-util apps get-reconcile-results command'

### PR DESCRIPTION
**Pull request by alexmt**
PR adds two argocd-util command that will help testing change implemented in https://github.com/argoproj/argo-cd/pull/3886

The commands allows to get app reconciliation status before deploying the new argocd version:

* get status before upgrading
```
argocd-util apps get-reconcile-results ./before-upgrade.yaml
```

* get status after upgrading
```
argocd-util apps get-reconcile-results --refresh ./after-upgrade.yaml
```

* print the difference

```
argocd-util apps diff-reconcile-results ./before-upgrade.yaml ./after-upgrade.yaml
```

Documentation will be added utility is tested internally and proved to be useful

### CODE REVIEW

File: controller/state.go [hunk 1]
```diff
@@ -81,6 +81,14 @@ type comparisonResult struct {
...
 }
 
+func (res *comparisonResult) GetSyncStatus() *v1alpha1.SyncStatus {
+	return res.syncStatus
+}
+
+func (res *comparisonResult) GetHealthStatus() *v1alpha1.HealthStatus {
+	return res.healthStatus
+}
+
```
Comment (jannfis) [inline controller/state.go:91]: Can these please have simple tests, so that we don't unnecessarily decrease test coverage?

**Review by jannfis — APPROVED** [review]
LGTM with just one minor comment.

File: cmd/argocd-util/commands/apps_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		},
+	)
+
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	assert.Equal(t, result[0].Health.Status, health.HealthStatusMissing)
+	assert.Equal(t, result[0].Sync.Status, v1alpha1.SyncStatusCodeOutOfSync)
+}
```
Comment (mayzhang2000) [inline cmd/argocd-util/commands/apps_test.go:119]: Looking great! Do we also need tests for `diffReconcileResults`?

Comment (alexmt) [reply cmd/argocd-util/commands/apps_test.go:119]: done

**Comment by alexmt** [comment]
@mayzhang2000 , @jannfis Thank you for review ! Added missing tests.

**Pull request merged**